src/schema/posts.ts

```diff
@@ -1521,6 +1677,63 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
 
       return { _: true };
     },
+    createSquadPostModeration: async (
+      _,
+      {
+        sourceId,
+        image,
+        title,
+        content,
+        type,
+        sharedPostId,
+      }: CreateSquadPostModerationArgs,
+      ctx: AuthContext,
+      info,
+    ): Promise<GQLSquadPostModeration> => {
```

Comment:
I think that we should not add an API endpoint for this. 
We should keep the create post endpoint for not change any logic on frontend post creation/share as [stated here](https://dailydotdev.atlassian.net/wiki/spaces/HAN/pages/1151500290/RFC-069+-+Admin+approval+on+Squads+posts#Post-creation.1)

Reply:
I think the core issue with the old one is when you create a freeform post without moderation, everything is as intended, but when you integrate moderation, the expected returned object would not be a post anymore. Either we return a union type or a `null` value. I am more inclined to the `null` value personally.

Reply:
@ilasw I discussed this with @sshanzel and we pretty much concluded what he wrote up here. I personally prefer not to return null because it isn't really clear why that is being returned and is quite flaky design wise.

I am not familiar with union types in graphql, but if we really want to keep it in the same endpoint, then I'd prefer to go that way.

Reply:
@capJavert Do you have any thoughts on this?

Reply:
Not 100% taking all logic into account, but I would rather have queries/mutations return the same thing always, oppose to union type which I agree would just complicate. Things like this are hard to visual in the RFC sometimes so changes in implementation sometimes do happen. 

So I am fine with leaving it as separate for now as long as we are able to implement all the other requirements.

Reply:
@sshanzel can you please point where I can find this logic in code? Thanks! 🙏

Reply:
PS: @sshanzel @AmarTrebinjac do you have calculated the potential effort impact of this change on frontend? If yes please can you add a section on RFC so we can track this activity? Thanks!

Reply:
> I would rather have queries/mutations return the same thing always, oppose to union type which I agree would just complicate (same for null).

I agree. It would be harder to catch when the form of the object is either A or B.

> do you have calculated the potential effort impact of this change on frontend?

In frontend it should be just a simple ternary operator on which to call whether the squad requires moderation or not. We should have something similar in the codebase. Same amount of parameters, same requirements, just a different endpoint.

Reply:
I have already mostly completed the frontend task for this !

Reply:
@sshanzel 
> In frontend it should be just a simple ternary operator

This calls for mistakes as well though, i can already envision we forget to add it in one specific place (let's say mobile via settings posting flow) and only realize weeks down the line.

In general it would be great if we could uniform some actions somehow?
I'm still ok with splitting, but maybe in FE it can be 1 call and that call determines in a hook what one to publish?

CC: @capJavert // @AmarTrebinjac would love to hear thoughts on how we can minimize risk with 2 flows.

Reply:
> This calls for mistakes as well though, i can already envision we forget to add it in one specific place (let's say mobile via settings posting flow) and only realize weeks down the line.

The condition should lie in a single place (a hook, ideally) where all decisions are made, not on the component level. Otherwise, it would be a mess to put conditions all over the place in each flow.